pkg/app/app.go

```diff
@@ -276,6 +276,38 @@ func (a *App) ConfigMapRefs() map[reftracker.RefKey]struct{} {
 	return configMaps
 }
 
+func (a *App) isSafeToFallbackToNoopDelete() bool {
+	if a.app.Status.Deploy == nil || a.app.Status.Deploy.KappDeployStatus == nil || a.app.Spec.Cluster != nil {
```

Comment:
One thing that I forgot last time was that the cluster can be non nil and we still might be relying on the SA if it is non nil as well, so we should be checking spec.SA instead.

Reply:
Ah I see, you mean we give preference to SA? So checking `a.app.ServiceAccountName == nil` is safer here?

Reply:
Made this change